oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/server/authentication/ServerBearerTokenAuthenticationConverterTests.java

```diff
@@ -217,6 +219,15 @@ void resolveWhenQueryParameterHasMultipleAccessTokensThenOAuth2AuthenticationExc
 
 	}
 
+	// gh-16038
+	@Test
+	void resoleWhenAllowUriQueryParameterIsFalseThenQueryParameterIsIgnored() {
```

Comment:
I think we would be missing a similar test to the ones you added in `DefaultBearerTokenResolverTests`, that ensure when two access tokens are present and `allowUriQueryParameter` is default (`false`) no exception is thrown?